lib/philomena/users.ex

```diff
@@ -943,7 +943,10 @@ defmodule Philomena.Users do
   defp setup_roles(nil), do: nil
 
   defp setup_roles(user) do
-    role_map = Map.new(user.roles, &{&1.resource_type || &1.name, &1.name})
+    role_map =
+      user.roles
+      |> Enum.group_by(& &1.resource_type, & &1.name)
```

Comment:
Previously the expression was `&1.resource_type || &1.name` meaning it was handling the case where `&1.resource_type` could be `nil`. I suppose it never is `nil`? Becase defaulting `resource_type` to `name` was strange. Just want to confirm that non-`nil` assumption in new code is correct.

Reply:
`resource_type` shouldn't be nil although it's missing a db constraint. I mentioned this in a cleanup pr.